packages/atomic/src/components/common/checkbox.spec.ts

```diff
@@ -0,0 +1,125 @@
+import {fireEvent, within} from '@storybook/test';
+import {html, render} from 'lit';
+import {vi} from 'vitest';
+import {checkbox, CheckboxProps} from './checkbox';
+
+describe('checkbox', () => {
+  let container: HTMLElement;
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+  });
+
+  const renderCheckbox = (props: Partial<CheckboxProps>): HTMLButtonElement => {
+    render(
+      html`${checkbox({
+        ...props,
+        checked: props.checked ?? false,
+        onToggle: props.onToggle ?? vi.fn(),
+      })}`,
+      container
+    );
+    return within(container).getByRole('checkbox') as HTMLButtonElement;
+  };
+
+  it('should render a checkbox with the correct text attributes', () => {
+    const props = {
+      text: 'Test Checkbox',
+      ariaLabel: 'Test Checkbox',
+    };
+
+    const button = renderCheckbox(props);
+
+    expect(button).toBeInTheDocument();
+    expect(button.getAttribute('aria-label')).toBe('Test Checkbox');
+    expect(button.value).toBe('Test Checkbox');
+  });
+
+  it('should not be checked by default', async () => {
+    const button = renderCheckbox({});
+
+    expect(button.getAttribute('aria-checked')).toBe('false');
+    expect(button.classList.contains('selected')).toBe(false);
+  });
+
+  it('should be have selected attributes and classes if checked', async () => {
```

Comment:
Should we also validate that the classList will *not* contain `selected` when `checked` is `false`?